src/components/clientTable/ClientTable.tsx

```diff
@@ -0,0 +1,114 @@
+"use client";
+import { useState } from "react";
+
+import {
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  getFilteredRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { Separator } from "@/components/ui/separator";
+
+interface DataTableProps<TData, TValue> {
+  columns: ColumnDef<TData, TValue>[];
```

Comment:
Is there a need for this level of abstraction? Did you intend for this table to be used elsewhere with custom columns? If you intended that this table is only used for the clients page, it would be worth removing this as a prop and using the columns defined in `src/components/clientTable/columns.tsx` directly in the `ClientTable` component body.